csrc/preseg_passes/mark_aliases_prepare.cpp

```diff
@@ -99,6 +100,16 @@ void insertSegmentSetAfter(
 
   // The general case.
   TensorView* copy = segment_set(use_of);
+  // Inherit the allocation domain from `use_of`. This is important to pass
+  // AliasTest.Bookend_SegmentSetPreservesAllocation.
+  TensorDomain* replayed_domain =
+      TransformReplay::replayCasP(
+          copy, use_of, -1, TransformReplayOptions().replayAllocation())
+          .first;
+  if (replayed_domain->hasAllocation()) {
+    copy->setAllocationDomain(
+        replayed_domain->allocation(), replayed_domain->contiguity());
+  }
```

Comment:
so previously when we forgot to update the allocation domain on the intermediate `copy` here, we ended up with two copies?!

Reply:
In some cases, yes. markAliases treats an empty allocation domain as default.